lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -65,14 +67,20 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
 
         # Retrieve the ES indices that belong to this run_id from the metadata
         # table
+        try:
+            indices = self.get_index(dataset, "run-toc")
+        except InvalidAPIRequest:
+            raise InvalidAPIRequest(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                "Indices from metadata table not found",
+            )
```

Comment:
Let's not do this. You're *catching* one `InvalidAPIRequest` that was raised where the problem was discovered and raising a *new* `InvalidAPIRequest` further from the problem. That's counter-productive. We should just let the original exception propagate out to the external handler unless we really need to

1. Add logger information to help triage,
2. Raise a different exception to better frame the problem at this level,
3. Perform local cleanup before letting the exception propagate

None of these are true here.